rich/syntax.py

```diff
@@ -286,30 +288,34 @@ def from_path(
             word_wrap (bool, optional): Enable word wrapping of code.
             background_color (str, optional): Optional background color, or None to use theme color. Defaults to None.
             indent_guides (bool, optional): Show indent guides. Defaults to False.
+            lexer_name (str, optional): Lexer name. Defaults to None.
 
         Returns:
             [Syntax]: A Syntax object that may be printed to the console
         """
-        with open(path, "rt", encoding=encoding) as code_file:
-            code = code_file.read()
+        if not path or path == "-":
```

Comment:
Not keen on this I'm afraid. The function is called `from_path` but this change means it might not use the path.

You can create a Syntax object from the regular constructor, which takes the `code` plus `lexer_name`.

Reply:
you're right, I dunno what I was thinking there! 

Anyway, I just pushed a change that's far less invasive and should work just as well.